lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3325,6 +3325,26 @@ namespace {
       return result;
     }
 
+    bool isCxxRecordUnimportable(const clang::CXXRecordDecl *decl) {
+      for (auto ctor : decl->ctors()) {
+        if (ctor->isCopyConstructor()) {
+          // If we have no way of copying the type we can't import the class
+          // at all because we cannot express the correct semantics as a swift
+          // struct.
+          if (ctor->isDeleted() || ctor->getAccess() != clang::AS_public)
+            return true;
+        }
+      }
+
+      if (auto dtor = decl->getDestructor()) {
+        if (dtor->isDeleted() || dtor->getAccess() != clang::AS_public) {
+          return true;
+        }
+      }
+
+      return false;
```

Comment:
Style note: This code might be a bit more readable if you instead had it like:

```
// destructor check
// if (auto dtor = ...) [...]
// constructor check
return llvm::any(..) // or llvm::all
```

That has 2 exits (instead of 3), and using `any`/`all` would express intent a bit more clearly compared to an explicit loop.

Reply:
You're right. This is a lot better. Thanks!